test/master-playlist-controller.test.js

```diff
@@ -1184,16 +1192,103 @@ QUnit.test('correctly sets alternate audio track kinds', function(assert) {
                'spanish track\'s kind is "alternative"');
 });
 
+QUnit.test('trigger events when an AES is detected', function(assert) {
+  let hlsaes = 0;
+  let playlistCopy = Hls.Playlist;
```

Comment:
You need to save a reference to the original `Hls.Playlist.isAes` function and restore it at the end instead of the `Hls.Playlist` object. since `Hls.Playlist` is an object, `playlistCopy` is just a reference to the object and not an actual copy. Same goes for `isFmp4`